src/ComServerHelpers/README.md

```diff
@@ -0,0 +1,136 @@
+2. Asynchronous methods must use the WinRT types (`IAsyncAction`,
+   `IAsyncActionWithProgress<TProgress>`, `IAsyncOperation<TResult>`,
+   `IAsyncOperationWithProgress<TResult, TProgress>`) instead of `Task` and
+   `Task<T>`.
+3. Event delegates must be either `TypedEventHandler<TSender, TResult>` or
+   `EventHandler<TResult>` instead of `EventHandler` and `EventHandler<T>`.
+4. Types in method parameters, type parameters, and return types must be:
+
+   * A blittable type.
+   * An interface that has a [.NET/WinRT Mapping](https://learn.microsoft.com/en-us/windows/apps/develop/platform/csharp-winrt/net-mappings-of-winrt-types).
+   * A WinRT type.
+   * Another interface in the project.
+
+5. Methods, properties, and events are all supported.
+
+## Metadata Project
+
+The metadata project is a C++/WinRT project that uses [MIDL 3.0](https://learn.microsoft.com/en-us/uwp/midl-3/) to create
+proxy types in a WinMD that can be referenced by the client of the OOP Server.
+No actual C++ code is needed, only the IDL.
+
+The IDL is very simple, only needing `runtimeclass`es that implement the
+interface from the contract project. Unlike in C#, in MIDL 3.0 the type
+automatically has the members from the interface so they do not need to be
+listed again. Importantly the `runtimeclass` must have an empty constructor,
+otherwise the proxy type cannot be created.
+
+> :exclamation:\*\*Important\*\*: Because of the mix of SDK Style and C++/WinRT,
+> `nuget restore` is needed to restore for C++/WinRT. In addition
+> `<RestoreProjectStyle>Packages.config</RestoreProjectStyle>` is needed in the
+> C++ project file.
+
+## Server Project
+
+The server project is the only project that references `ComServerHelpers`.
+It will contain implementations of the interfaces from the contract and when run
+should register them with an instance of `COMServer` for COM activation and
+`WinRtServer` for WinRT activation. The implementations must have a GUID using
+the `System.Runtime.InteropServices.GuidAttribute` attribute.
+
+Because the interfaces must use the WinRT asynchronous types instead of the .NET
+ones, the implementation will likely need to use `AsyncInfo` to help adapt
+between the two systems.
+
+## Client Project
+
+A client can be both full trust applications (Win32, WPF, WinForms, etc) or a
+UWP app.
+
+A UWP client cannot use WinRT activation and must use COM style activation. The
+UWP sample app shows how to do this. To understand the details behind it, see
+[this blog post](https://devblogs.microsoft.com/ifdef-windows/the-journey-of-moving-from-cpp-winrt-to-csharp-in-the-microsoft-store/).
+
+A full trust client can use WinRT activation, which allows you to create the
+remote instances simply by `new SomeType()`, like you would for any other type.
+The sample WPF application shows this in action (using with WinForms would be
+similar).
+
+# Sample
+
+To help understand usage and show what can be done samples can be found under
```

Comment:
I don't believe these were brought over right?  Doesn't need to happen in this PR, but we should add some of these samples or equivalent and also use them as tests that we build and run or so.